modules/navier_stokes/test/tests/finite_element/ins/lid_driven/steady_vector_fsp_al.i

```diff
@@ -0,0 +1,165 @@
+rho=1
+mu=1e-3
+U=1
+l=1
+prefactor=${fparse 1/(l/2)^2}
+n=8
+gamma=${U}
+
+[Mesh]
+  [gen]
+    type = GeneratedMeshGenerator
+    dim = 2
+    xmin = 0
+    xmax = ${l}
+    ymin = 0
+    ymax = ${l}
+    nx = ${n}
+    ny = ${n}
+    elem_type = QUAD4
+  []
+  second_order = true
+[]
+
+[Variables]
+  [vel]
+    order = SECOND
+    family = LAGRANGE_VEC
+  []
+  [p]
+    order = FIRST
+    family = LAGRANGE
+  []
+[]
+
+[Kernels]
+  [mass]
+    type = INSADMass
+    variable = p
+  []
+  [mass_kernel]
+    type = MassMatrix
+    variable = p
+    matrix_tags = 'mass'
+    density = ${fparse -gamma - mu}
+  []
+  [momentum_advection]
+    type = INSADMomentumAdvection
+    variable = vel
+  []
+  [momentum_viscous]
+    type = INSADMomentumViscous
+    variable = vel
+  []
+  [momentum_pressure]
+    type = INSADMomentumPressure
+    variable = vel
+    pressure = p
+    integrate_p_by_parts = true
+  []
+  [momentum_graddiv]
+    type = INSADMomentumGradDiv
+    variable = vel
+    gamma = ${gamma}
+  []
```

Comment:
should this contribute to the residual too?
I guess I ll know once I read the paper but I was expecting it from the doco to only contribute to the mass matrix

Reply:
yup I see it equation 2.6 of Farrell